statsmodels/discrete/discrete_model.py

```diff
@@ -2371,11 +2371,10 @@ def __init__(self, model, mlefit, cov_type='nonrobust', cov_kwds=None,
 
 
     def __getstate__(self):
-        try:
-            #remove unpicklable callback
-            self.mle_settings['callback'] = None
-        except (AttributeError, KeyError):
-            pass
+        # remove unpicklable methods
+        self.mle_settings['callback'] = None
```

Comment:
This removes the try ... except protection for the case when 'callback' is not available. 
It doesn't sound like a required key.

Reply:
Better safe than sorry.
I think I want to keep both `mle_settings` and 'callback' optional. Two extra if won't change performance.
e.g. we, or somebody, might replace the fit method, e.g. if we use GLMNet type of coordinate descend.